share/python/procedures/findif_response_utils/tensor_helper.py

```diff
@@ -0,0 +1,118 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2016 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+"""
+Module of helper functions for distributed ccresponse computations.
+
+Defines functions for retrieving tensors computed at displaced geometries.
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+import collections
+import shelve
+import copy
+import os
+import psi4
+import p4util
+from p4const import *
+
+
+def synthesize_displaced_tensor(db, signature, row_dim):
```

Comment:
@amjames Ah, I didnt know more `findif` python side was coming. This is great! In that case we will probably move this up a few levels eventually, but this is a good place for now.

I like the alternative naming schemes and your right, its not actually a Psi4 Matrix and its good to be careful here. There was much discussion that smaller arrays such as geometries should be NumPy arrays due to the large flexibility that NumPy provides and the high level of interoperability between NumPy and Psi4 at the moment.

The proposed changes look good.